File: Paste as MD Link.lbaction/Contents/Scripts/default.js
// LaunchBar Action Script
function run(argument) {
    // Get URL from the clipboard
    url = LaunchBar.getClipboardString()
    // Check if it's valid URL
    function isValidUrl(url) {
        var pattern = new RegExp('^(https?:\\/\\/)?' + // protocol
            '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
            '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
            '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
            '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
            '(\\#[-a-z\\d_]*)?$', 'i'); // fragment locator
        return !!pattern.test(url);
    }
    // Get title from the URL
    try {
        if (!isValidUrl(url)) throw new Error(url + ' is not a URL.');
        if (!/^https?:\/\//i.test(url)) url = 'http://' + url;

        LaunchBar.displayNotification({
            title: "Paste as MD Link",
            string: "Request title from " + url
        });
        LaunchBar.debugLog('Start requesting.');
        var result = HTTP.loadRequest(url, {
            timeout: 8,
            method: 'GET',
            resultType: 'text'
        });
        LaunchBar.debugLog('Finish requesting.');
        LaunchBar.debugLog("Result: ", Object.keys(result));
        if (result.data != undefined) {
            LaunchBar.debugLog(result.data);
            LaunchBar.debugLog("Start matching.");
            const matches = result.data.match(/<head[\s\S]*?>[\s\S]*?<title[\s\S]*?>\s*(.*)\s*<\/title>[\s\S]*?<\/head>/);
            LaunchBar.debugLog("Finish matching.");
            // LaunchBar.log('matches', matches[1]);
            const title = matches[1];
            const mdLink = '[' + title + '](' + url + ')';
            LaunchBar.debugLog('mdLink', mdLink);
            // LaunchBar.displayNotification({
            //     title: "Paste as MD Link",
            //     string: "Success!"
            // });
            LaunchBar.paste(mdLink);
        } else if (result.timedOut) {
            LaunchBar.displayNotification({
                title: "Paste as MD Link",
                string: "Error: " + 'Unable to load the LaunchBar page due to a timeout'
            });
        } else if (result.error != undefined) {
            LaunchBar.displayNotification({
                title: "Paste as MD Link",
                string: "Error: " + 'Unable to load the LaunchBar page:' + result.error
            });
        }
    } catch (e) {
        LaunchBar.displayNotification({
            title: "Paste as MD Link",
            string: "" + e
        });
        LaunchBar.debugLog('' + e);
    }
}

File: Paste as MD Link.lbaction/Contents/Scripts/default.py
#!/usr/local/Caskroom/miniconda/base/bin/python
#
# LaunchBar Action Script
#

from AppKit import NSPasteboard, NSStringPboardType
import pyperclip
import os
import requests
import re
from bs4 import BeautifulSoup
from urllib.parse import urlparse

def isPrefixHTTP(url):
    return re.match(re.compile(r'^https?:\/\/', re.IGNORECASE), url) is not None
def isValidURL(url):
    pattern = re.compile(
        r'^(?:http|ftp)s?://' # http:// or https://
        r'(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|' #domain...
        r'localhost|' #localhost...
        r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})' # ...or ip
        r'(?::\d+)?' # optional port
        r'(?:/?|[/?]\S+)$', re.IGNORECASE)
    return re.match(pattern, url) is not None
def notify(title, text):
    os.system("""
              osascript -e 'display notification "{}" with title "{}"'
              """.format(text, title))
def ospaste():
    os.system("""
              osascript -e 'tell application "System Events" to keystroke "v" using command down'
              """)

pb = NSPasteboard.generalPasteboard()
url = pb.stringForType_(NSStringPboardType)

notify('LaunchBar: Paste as MD Link', 'Requesting: ' + url)

try:
    if not isPrefixHTTP(url): url = 'https://' + url
    if not isValidURL(url):
        raise Exception(url + ' is not a valid URL.')
    else:
        req = requests.get(url)
        if req.status_code == 200:
            soup = BeautifulSoup(req.text)
            title = soup.head.title.text
            mdlink = '[' + title + '](' + url + ')'
            notify('LaunchBar: Paste as MD Link', 'MDLink: ' + mdlink)
            pyperclip.copy(mdlink)
            ospaste()
        else:
            raise Exception('problem with request: ', req.status_code)
except Exception as e:
    notify('LaunchBar: Paste as MD Link', 'Error:' + e.args[0])